services/graph/pkg/service/v0/utils.go

```diff
@@ -454,3 +458,19 @@ func roleConditionForResourceType(ri *storageprovider.ResourceInfo) (string, err
 		return "", errorcode.New(errorcode.InvalidRequest, "unsupported resource type")
 	}
 }
+
+// GetShareID returns a valid share ID from the given resource ID.
+// Basically, it just checks if the given ID has an opaque ID that is
+// compatible with the id format defined by storagespace package.
+func GetShareID(shareID storageprovider.ResourceId) (*collaboration.ShareId, error) {
+	if _, err := storagespace.ParseID(shareID.GetOpaqueId()); err != nil {
```

Comment:
I am not sure what this is supposed to prove. The fact that the opaqueId itself parses as a resource id. Does not prove that it is a valid share id. I think this is misleading.

Reply:
yes true, but doesnt it make sense to have it and be sure that the opaque has the correct format?